### ISSUE TITLE
Mark PDF as important 

### ISSUE THREAD

**Issue by amarradi** [michaeltroger/greenpass-android#269]
Is it possible to insert an additional tag to mark important documents? I currently have about 20 documents in the database, not a lot but I have lost track of the invoice for the tourist tax from Novasol. An idea I have is some kind of star or other symbol to highlight the PDF. Is that possible? I would add a boolean variable to the data structure and either set it or not

**Comment by michaeltroger** [comment]
Thx for the suggestion. Generally speaking I can imagine this, however I would not be sure how the user interface should look like for the 2 different layouts (side-by-side vs. list layout).
Some challenges would also be the "insert to front" setting and the "change of document order" option.
From the data structure it would be easy to add, I agree, but I believe it requires some thinking and effort to build a meaningful user interface.

**Comment by amarradi** [comment]
In the list view, I would set the file renaming function to the setOnLongClick and at this position I would set the new "icon" for important. It is also possible to set the new icon to the setOnLongClick

![listview-idea.png](https://github.com/michaeltroger/greenpass-android/assets/720573/23388cfd-d4f1-4c2e-aec4-966f57343a36)



In the "Grid" layout, I would place the new icon directly above it 

![gridview.idea.png](https://github.com/michaeltroger/greenpass-android/assets/720573/b756b2ec-f220-4e65-b43b-b1b6e7e8c4f3)

**Comment by michaeltroger** [comment]
I don't think "long press" to rename would be obvious to users. A common UX for Android apps is that long press will display all operations related to the file (like renaming, sharing, deleting) in the toolbar. That's something I could imagine.
Also delete and sharing icons could be replaced by gestures - swiping left and right. Then there would be plenty of space for the star icon. 
A remaining question would be if gestures or long press would be obvious for users or if an onboarding tooltip would be required. Currently I think I would prefer the long press approach to show all operations.

For the "grid layout" you mean the icon would be next to the delete and share icons? Or above them? For this one I'm not sure yet how I could improve the UI for another icon like the star.

Also I still believe favoriting a document could e.g. clash with the ordering functionality if not implemented wisely.

**Comment by amarradi** [comment]
You're right, this still needs to be implemented very well and I haven't even thought about sorting yet. If you only set the favorites star in the document and not in the list or "grid view", you can sort the list or grid after closing the open document. This way you don't have a problem with the view as you don't have to change anything.

I would add the star to the left side of the recycle bin

**Comment by amarradi** [comment]
Do you notally App. Its an easy Note App. At this app you can mark documents by long pressing and in the document. At long pressing the app bar changed and you see a pin needle
